packages/server/src/fhir/search.test.ts

```diff
@@ -2906,6 +2906,39 @@ describe('FHIR Search', () => {
         expect(result.entry?.[0]?.resource?.id).toEqual(patient.id);
       }));
 
+    test('_filter with chained search', () =>
+      withTestContext(async () => {
+        const mrn = randomUUID();
+        const patient = await repo.createResource<Patient>({
+          resourceType: 'Patient',
+          name: [{ given: ['Eve'] }],
+          identifier: [{ system: 'http://example.com/mrn', value: mrn }],
+        });
+
+        const observation = await repo.createResource<Observation>({
+          resourceType: 'Observation',
+          status: 'final',
+          code: {
+            text: 'Strep test',
+          },
+          subject: createReference(patient),
+        });
+
+        const result = await repo.search({
+          resourceType: 'Observation',
+          filters: [
+            {
+              code: '_filter',
+              operator: Operator.EQUALS,
+              value: `subject:Patient.identifier eq http://example.com/mrn|${mrn}`,
+            },
+          ],
+        });
+
+        expect(result.entry).toHaveLength(1);
+        expect(result.entry?.[0]?.resource?.id).toEqual(observation.id);
+      }));
+
```

Comment:
The canonical use case of this is to use `or`, so it's with having a test case for this. 

From [this thread](https://medplum.slack.com/archives/C04A55B3VU5/p1715703564822689):
"I want to search Tasks where the `Task.patient` has an identifier with system `http://example.com|123` OR the patient has no identifier with system `http://example.com`